sycl/include/CL/sycl/handler.hpp

```diff
@@ -1617,6 +1617,22 @@ class __SYCL_EXPORT handler {
     // for the device.
     size_t MaxWGSize =
         ext::oneapi::detail::reduGetMaxWGSize(MQueue, OneElemSize);
+
+    // The maximum WGSize returned by CPU devices is very large and does not
+    // help the reduction implementation: since all work associated with a
+    // work-group is typically assigned to one CPU thread, selecting a large
+    // work-group size unnecessarily increases the number of accumulators.
+    // The default of 16 was chosen based on empirical benchmarking results;
+    // an environment variable is provided to allow users to override this
+    // behavior.
+    if (detail::getDeviceFromHandler(*this).is_cpu()) {
+      if (const char *MaxWGSizeEnv = getenv("SYCL_CPU_REDUCTION_MAX_WG_SIZE")) {
```

Comment:
Do we need to update a file documenting other env variables with this one's description?

Also, I'm not fluent with the codebase, but so far all the settings I've seen were done using static variables so that the value is read exactly one.  How common is the approach taken here?

Reply:
Yeah, absolutely. I have changed the name of the environment variable slightly, added a GPU and ACC variant, defined them as `SYCLConfig`, and documented them in sycl/doc/EnvironmentVariables.md.

I have also added some additional restrictions on the environment variable, such as it throwing an error if it is invalid and if it exceeds the maximum on the device that will be used instead. The checks have also been moved to `reduGetMaxWGSize`, partly because configs needs to be in source and partly because it seems to be the right place to overwrite the maximum work groups side (also reducing unneeded work.)